# Project 4 -- Simrith Ranjan

**TA Help:** 
    
**Collaboration:** Anjali

provided me motivation and guidance when estimating how long each question would take

## Question 1

In [1]:
x <- 7
if (x > 0) {
    print("Positive Number")
}

[1] "Positive Number"


In [2]:
options(jupter.rich_display = F)

In [3]:
options(repr.matrix.max.cols = 26, repr.matrix.max.rows = 200)

In [34]:
myDF <- read.csv("/anvil/projects/tdm/data/craigslist/vehicles.csv")

In [35]:
head(myDF, n = 50)

id        
1  7222695916
2  7218891961
3  7221797935
4  7222270760
5  7210384030
6  7222379453
7  7221952215
8  7220195662
9  7209064557
10 7219485069
11 7218893038
12 7218325704
13 7217788283
14 7217147606
15 7209027818
16 7223509794
17 7222753076
18 7222206015
19 7220030122
20 7218423006
21 7216672204
22 7215617048
23 7213839225
24 7208549803
25 7213843538
26 7212631321
27 7219973522
28 7316814884
29 7316814758
30 7316814989
31 7316743432
32 7316356412
33 7316343444
34 7316304717
35 7316285779
36 7316257769
37 7316133914
38 7316130053
39 7315816316
40 7315770394
41 7315756348
42 7315715960
43 7315379459
44 7315275975
45 7315275942
46 7315270785
47 7315039307
48 7314910156
49 7314854462
50 7314811916
   url                                                                                            
1  https://prescott.craigslist.org/cto/d/prescott-2010-ford-ranger/7222695916.html                
2  https://fayar.craigslist.org/ctd/d/bentonville-2017-hyundai-elantra-se/7218891961.html         
3  https://keys.craigslist.org/cto/d/summerland-key-2005-excursion/7221797935.html                
4  https://worcester.craigslist.org/cto/d/west-brookfield-2002-honda-odyssey-ex/7222270760.html   
5  https://greensboro.craigslist.org/cto/d/trinity-1965-chevrolet-truck/7210384030.html           
6  https://hudsonvalley.craigslist.org/cto/d/westtown-2007-ford-150/7222379453.html               
7  https://hudsonvalley.craigslist.org/cto/d/westtown-silverado-2000/7221952215.html              
8  https://hudsonvalley.craigslist.org/cto/d/poughquag-2015-acura-rdx-warranty/7220195662.html    
9  https://medford.craigslist.org/cto/d/grants-pass-two-2002-bmw-tii/7209064557.html              
10 https://erie.craigslist.org/cto/d/erie-2012-suzuki-sx4/7219485069.html                         
11 https://elpaso.craigslist.org/ctd/d/el-paso-2011-chevy-chevrolet-malibu-ls/7218893038.html     
12 https://elpaso.craigslist.org/ctd/d/el-paso-2017-ford-fusion-platinum-sedan/7218325704.html    
13 https://elpaso.craigslist.org/ctd/d/el-paso-2013-ford-focus-se-sedan/7217788283.html           
14 https://elpaso.craigslist.org/ctd/d/el-paso-2019-toyota-4runner-limited/7217147606.html        
15 https://elpaso.craigslist.org/ctd/d/el-paso-2012-buick-enclave-convenience/7209027818.html     
16 https://bellingham.craigslist.org/ctd/d/bellingham-2007-mercedes-benz-slk-slk/7223509794.html  
17 https://bellingham.craigslist.org/ctd/d/bellingham-2015-mercedes-benz-class-400/7222753076.html
18 https://bellingham.craigslist.org/ctd/d/bellingham-2016-bmw-x1-xdrive28i/7222206015.html       
19 https://bellingham.craigslist.org/ctd/d/bellingham-2017-bmw-x3-xdrive28i/7220030122.html       
20 https://bellingham.craigslist.org/ctd/d/bellingham-2011-mercedes-benz-class-300/7218423006.html
21 https://bellingham.craigslist.org/ctd/d/bellingham-2015-mercedes-benz-class-400/7216672204.html
22 https://bellingham.craigslist.org/ctd/d/bellingham-2016-bmw-x1-xdrive28i/7215617048.html       
23 https://bellingham.craigslist.org/ctd/d/bellingham-2017-bmw-x3-xdrive28i/7213839225.html       
24 https://bellingham.craigslist.org/ctd/d/bellingham-2011-mercedes-benz-class-300/7208549803.html
25 https://skagit.craigslist.org/ctd/d/bellingham-2015-mercedes-benz-class-400/7213843538.html    
26 https://skagit.craigslist.org/ctd/d/bellingham-2016-bmw-x1-xdrive28i/7212631321.html           
27 https://lacrosse.craigslist.org/cto/d/la-crosse-01-pontiac-sunfire/7219973522.html             
28 https://auburn.craigslist.org/ctd/d/auburn-university-2014-gmc-sierra-1500/7316814884.html     
29 https://auburn.craigslist.org/ctd/d/auburn-university-2010-chevy-chevrolet/7316814758.html     
30 https://auburn.craigslist.org/ctd/d/auburn-university-2020-chevy-chevrolet/7316814989.html     
31 https://auburn.craigslist.org/ctd/d/auburn-university-2017-toyota-tundra/7316743432.html       
32 https://auburn.craigslist.org/cto/d/auburn-university-2013-150/7316356412.html                 
33 https://auburn.crai

In [6]:
dim(myDF)

[1] 426880     26

In [7]:
myDF$texasflag <- rep("not Texas", times = 426880)

In [10]:
myDF$texasflag[myDF$state == "tx"] <- "yes Texas"

In [13]:
dim(subset(myDF, state == "tx"))

[1] 22945    27

In [14]:
dim(subset(myDF, texasflag == "yes Texas"))

[1] 22945    27

In [20]:
myDF$pricecategory <- rep("cheap car", times = 426880)

In [21]:
myDF$pricecategory[myDF$price > 10000] <- "expensive car"

In [22]:
myDF$newyorkcars <- rep(FALSE, times = 426880)

In [23]:
myDF$newyorkers[myDF$state == "ny"] <- TRUE

In [24]:
myDF$pricecategory[myDF$price > 15000] <- "expensive car"

In [28]:
head(myDF$state)

[1] "az" "ar" "fl" "ma" "nc" "ny"

In [29]:
class(myDF$state)

[1] "character"

In [30]:
length(table(myDF$state))

[1] 51

In [32]:
head(table(myDF$state))


   ak    al    ar    az    ca    co 
 3474  4955  4038  8679 50614 11088 

In [36]:
myDF$price2000 <- rep(FALSE, times = 426880)

In [37]:
myDF$price2000[myDF$price >= 2000] <- TRUE

In [38]:
dim(subset(myDF, price2000 == TRUE))

[1] 373903     27

In [39]:
mean(myDF$price)

[1] 75199.03

How many unique states are there in total? Which five of the states have the most occurrences? 51 states. Five states with the most occurences is alaska, alabama, arkansas, arizona, california, colorado

How many cars have a price that is greater than or equal to $2000 ? 373903 cars have a price that is greater than or equal to 2000.

What is the average price of the vehicles in the dataset?
$75199.03 is the average price of the vehicles in this dataset. (which is a lot, but on craigslist they'll be very large values and silly values that throws off the average)

## Question 2

In [43]:
myDF$pricecategory <- cut(myDF$price, breaks = c(-Inf, 10000, 15000, Inf), labels = c("cheap", "expensive", "very very expensive"))

In [18]:
mileage_category <- cut(myDF$odometer, breaks = c(-Inf, 50000, 100000, 150000, Inf), labels = c("low", "moderate", "high", "very high"))

In [23]:
has_VIN <- rep("has vin", times = 426880)

In [5]:
myDF$has_VIN[myDF$id != "NA"] <- "no vin"

In [6]:
mynchar <- nchar(as.character(myDF$description))

In [7]:
head(mynchar, n = 50)

[1]    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
[16]    0    0    0    0    0    0    0    0    0    0    0    0 4368 4581 4528
[31] 4684   66 4441 4553 1201 4503 4336  129 5031 4579 4433 4616 4397 4585 4359
[46] 4716  726 4967 4615 4673

In [25]:
description_length <- cut(mynchar, breaks = c(0, 50, 100, 200, 500, Inf), labels = c("very short", "short", "medium", "long", "very long"))

In [13]:
head(myDF$description_length)

[1] <NA> <NA> <NA> <NA> <NA> <NA>
Levels: very short short medium long very long

## Question 3

In [21]:
table(mileage_category)

mileage_category
      low  moderate      high very high 
   135454    111687     99102     76237 

low - 135454
moderate - 111687
high - 99102
very high - 76237

In [24]:
table(has_VIN)

has_VIN
has vin 
 426880 

has vin - 426880

In [26]:
table(description_length)

description_length
very short      short     medium       long  very long 
      2933       7688      24967      60718     330505 

very short - 2933
short - 7688
medium - 24967
long - 60718
very long - 330505

## Question 4

In [27]:
myTexasDF <- subset(myDF, state == "tx")

In [30]:
myAkDF <- subset(myDF, state == "ak")

In [31]:
myVaDF <- subset(myDF, state == "va")

## Question 5

In [33]:
myTexasDF <- subset(myDF, state == "tx" & !is.na(lat) & !is.na(long))

In [38]:
library(leaflet)

In [39]:
library(sf)

In [42]:
points <- st_as_sf(myTexasDF, coords=c("long", "lat"), crs=4326)

In [45]:
addCircleMarkers(addTiles(leaflet(myTexasDF)), radius = 1)

Assuming "long" and "lat" are longitude and latitude, respectively



HTML widgets cannot be represented in plain text (need html)

In [46]:
addCircleMarkers(addTiles(leaflet(myAkDF)), radius = 1)

Assuming "long" and "lat" are longitude and latitude, respectively

Warning message in validateCoords(lng, lat, funcName):
"Data contains 1 rows with either missing or invalid lat/lon values and will be ignored"


HTML widgets cannot be represented in plain text (need html)

In [47]:
addCircleMarkers(addTiles(leaflet(myVaDF)), radius = 1)

Assuming "long" and "lat" are longitude and latitude, respectively

Warning message in validateCoords(lng, lat, funcName):
"Data contains 973 rows with either missing or invalid lat/lon values and will be ignored"


HTML widgets cannot be represented in plain text (need html)

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.